version: "3.9"
services:
  app:
    build:
      context: .
      dockerfile: ./docker/app/Dockerfile
      target: dev
    ports:
      - 8080:8080
    volumes:
      - .:/app
    tty: true
    depends_on:
      db:
        condition: service_healthy
    working_dir: /app

  db:
    image: mysql:8.0
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_USER: test
      MYSQL_PASSWORD: test
      MYSQL_DATABASE: test
      TZ: Asia/Tokyo
    ports:
      - 3306:3306
    volumes:
      - auth-db-data:/var/lib/mysql
      - ./docker/db/init.d:/docker-entrypoint-initdb.d
      - ./docker/db/conf.d:/etc/mysql/conf.d
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost", "-u", "root"]
      interval: 5s
      timeout: 5s
      start_period: 5s
      retries: 3

volumes:
  auth-db-data:
